data/abilities.ts

```diff
@@ -1443,6 +1450,12 @@ export const Abilities: import('../sim/dex-abilities').AbilityDataTable = {
 				pokemon.formeChange(forme, this.effect, false, '0', '[msg]');
 			}
 		},
+		onEnd(pokemon) {
+			if (pokemon.baseSpecies.baseSpecies !== 'Castform' || pokemon.transformed) return;
+			if (pokemon.baseSpecies.id !== 'castform') {
```

Comment:
```suggestion
			if (pokemon.species.id !== 'castform') {
```
Also both abilities do check for `isActive` but in slightly different ways. I don't know why one checks for `hp` and the other does not though.

Reply:
It does, Castform hp check was at the end.